phoenix-core/src/main/java/org/apache/phoenix/compile/OffsetCompiler.java

```diff
@@ -62,23 +82,233 @@ public SortOrder getSortOrder() {
+                LOGGER.warn("Unexpected error while compiling RVC Offset, expected AndExpression got " + whereExpression.getClass().getName());
+                throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must specify the tables PKs.");
+            }
+
+            List<RowKeyColumnExpression> rowKeyColumnExpressionList = buildListOfRowKeyColumnExpressions(whereExpression.getChildren(),isIndex);
+            if(rowKeyColumnExpressionList.size() != numUserColumns) {
+                LOGGER.warn("Unexpected error while compiling RVC Offset, expected " + numUserColumns + " found " + rowKeyColumnExpressionList.size());
+                throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must specify the table's PKs.");
+            }
+
+            for (int i = 0; i < numUserColumns; i++) {
+                PColumn column = columns.get(i + userColumnIndex);
+                //
+                ColumnParseNode columnParseNode = rvcColumnParseNodeList.get(i);
+
+                String columnParseNodeString = columnParseNode.getFullName();
+                if (isIndex) {
+                    columnParseNodeString = IndexUtil.getDataColumnName(columnParseNodeString);
+                }
+
+                RowKeyColumnExpression rowKeyColumnExpression = rowKeyColumnExpressionList.get(i);
+                String expressionName = rowKeyColumnExpression.getName();
+
+                // Not sure why it is getting quoted
+                expressionName = expressionName.replace("\"", "");
+
+                if (isIndex) {
+                    expressionName = IndexUtil.getDataColumnName(expressionName);
+                }
+
+                if (!StringUtils.equals(expressionName, columnParseNodeString)) {
+                    throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must specify the tables PKs.");
+                }
+
+                //
+                String columnString = column.getName().getString();
+                if (isIndex) {
+                    columnString = IndexUtil.getDataColumnName(columnString);
+                }
+                if (!StringUtils.equals(expressionName, columnString)) {
+                    throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must specify the tables PKs.");
+                }
+            }
+
+            // check to see if this was a single key expression
+            ScanRanges scanRanges = context.getScanRanges();
+
+            if (!scanRanges.isPointLookup()) {
+                throw new RowValueConstructorOffsetNotCoercibleException("RVC Offset must be a point lookup.");
+            }
+
+            // Note the use of getLowerRange as this already handles the case of inclusionary offset
+            CompiledOffset compiledOffset = new CompiledOffset(Optional.<Integer>absent(),
+                    Optional.of(scanRanges.getScanRange().getLowerRange()));
+
+            return compiledOffset;
+        }
```

Comment:
We definitely should break this up into multiple methods and add unit tests 😄

Reply:
What set of breakup where you thinking?  I did add 2 methods which generate the lists to be verified.  Where you thinking to break out both initial and final verification into their own methods?